manager/includes/extenders/dbapi.mysqli.class.inc.php

```diff
@@ -223,27 +223,29 @@ public function query($sql, $watchError = true)
     public function delete($from, $where = '', $orderBy = '', $limit = '')
     {
         $modx = evolutionCMS();
-        $out = false;
         if (!$from) {
             $modx->messageQuit("Empty \$from parameters in DBAPI::delete().");
-        } else {
-            $from = $this->replaceFullTableName($from);
-            $where = trim($where);
-            $orderBy = trim($orderBy);
-            $limit = trim($limit);
-            if ($where !== '' && stripos($where, 'WHERE') !== 0) {
-                $where = "WHERE {$where}";
-            }
-            if ($orderBy !== '' && stripos($orderBy, 'ORDER BY') !== 0) {
-                $orderBy = "ORDER BY {$orderBy}";
-            }
-            if ($limit !== '' && stripos($limit, 'LIMIT') !== 0) {
-                $limit = "LIMIT {$limit}";
-            }
+            return false;
+        }
 
-            $out = $this->query("DELETE FROM {$from} {$where} {$orderBy} {$limit}");
+        $where = trim($where);
+        if(!$where && !$limit) {
+            return $this->truncate($from);
```

Comment:
Deleting all values is not the same as truncate, because truncate resets auto incremet counter

Reply:
@Pathologic 

Yes, of course I know it. In such a case, I thought it would be better to reset the ai number, but if it is better to leave it, I will do so.

Reply:
It would be better to leave as is to avoid unexpected behavior.

Reply:
ok, I will fix it soon.